operator-common/src/main/java/io/strimzi/operator/cluster/model/StatusDiff.java

```diff
@@ -24,7 +25,7 @@
     private final boolean isEmpty;
 
     public StatusDiff(Status current, Status desired) {
-        JsonNode source = patchMapper().valueToTree(current == null ? "{}" : current);
+        JsonNode source = patchMapper().configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true).valueToTree(current == null ? "{}" : current);
         JsonNode target = patchMapper().valueToTree(desired == null ? "{}" : desired);
```

Comment:
It does seems strange to me that the processing of the `current` and `desired` statuses is not the same before the comparison.

Reply:
I'm in agreement here, @stanlyDoge is there a reason line 29 is not as below?
``` 
JsonNode target = patchMapper().configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true).valueToTree(desired == null ? "{}" : desired);
```

It would make sense to me that we should be sorting both the source and the target for comparison?